### ISSUE TITLE
Action items for adoption of Model Registry in Kubeflow

### ISSUE THREAD

**Issue by rareddy** [kubeflow/community#685]
These are the action item that needs to be accomplished for Model Registry adoption in the Kubeflow community

Related issues: 

- https://github.com/kubeflow/kubeflow/issues/7396
- https://github.com/kubeflow/community/pull/682
- https://github.com/kubeflow/community/pull/673 

Below are the action items needed for the Model Registry

- [x]  Get approval from the Kubeflow Steering Committee to accept the development of Model Registry as a Kubeflow community component.
- [x] Model Registry Proposal submission
- [ ] Create a GitHub repository for Model Registry under Kubeflow organization
- [ ] Configure `tide` for the new repository
- [ ] Add Model Registry maintainers defined in the proposal to Kubeflow GitHub org https://github.com/kubeflow/internal-acls?tab=readme-ov-file#joining-kubeflow-github-organization
- [ ] Create OWNERS file with maintainers in the proposal
- [ ] Create a README file, which calls out 
    - Red Hat drives the project's development through Open Source principles, ensuring transparency, sustainability, and community ownership.
    - Red Hat values the Kubeflow community and commits to providing a minimum of 12 months' notice before ending project maintenance after the initial release.
- [ ] Find customer or user Zero to aid in the development of the Model Registry solution
- [ ] Define a roadmap for the Model Registry as to when it will be introduced into the Kubeflow release stream.

Please let me know if I missed something @jbottum @james-jwu @zijianjoy.

**Comment by zijianjoy** [comment]
Hello, following up on this topic.

Before creating repo for model registry in GitHub, we are looking for the following information to be in-place:
```
- [ ] Create a README file, which calls out 
    - Red Hat drives the project's development through Open Source principles, ensuring transparency, sustainability, and community ownership.
    - Red Hat values the Kubeflow community and commits to providing a minimum of 12 months' notice before ending project maintenance after the initial release.
- [ ] Find customer or user Zero to aid in the development of the Model Registry solution
```

Once these information are provided, I can help setting up a repo for model registry. For reference, here are the steps I will take: https://github.com/kubeflow/community/blob/master/repository-setup.md

**Comment by thesuperzapper** [comment]
Also, I must have missed it, but who are the non-IBM (including Red Hat) employees that are going to be maintainers/OWNERS?

I only see IBM employees in the [initial proposal](https://docs.google.com/document/d/1T3KfOqIfJohp0s1koQ2XrJJQQhj7TECO-m2xPsW59_c/edit#heading=h.t7b42oqmiz2y), and followup PR (https://github.com/kubeflow/community/pull/682).

**Comment by rareddy** [comment]
@thesuperzapper we are hoping to pick up community contributors with this effort. Since we are not an existing project and a new effort we do not have additional contributors other than those already defined. But we sure hope to change this soon. 

@zijianjoy Do you want us to provide a Readme attached to PR? I was thinking we will submit as PR on the repo. BTW, we are also working on confirming user zero, this will require some time on this.

IMO, there is nothing to lose from Kubeflow perspective until Model Registry gets included/distributed with Kubeflow distribution. So, I request some flexibility if you can meanwhile, while we can contribute to this to find contributors and users.

**Comment by james-jwu** [comment]
@zijianjoy I am okay with creating the readme file after creating the repo, and not making customer-zero a prerequisite. Let's proceed with the repo creation.

**Comment by zijianjoy** [comment]
@rareddy Please provide the following information in order to create repo:
1. Creation of the team who will be owner of the repo. Follow example in https://github.com/kubeflow/internal-acls/blob/5be405480241e2fc79f09c8462eb9fe69b656d24/github-orgs/kubeflow/org.yaml#L1024-L1035C19
1. Official name of the repo
2. Select a License for the project.

**Comment by rareddy** [comment]
@zijianjoy should I also add the owners of the repo as members at the beginning of the file in 1? or not needed?

**Comment by rareddy** [comment]
1) https://github.com/kubeflow/internal-acls/pull/642
2) model-registry
3) ASL 2.0

OWNERS
```
approvers:
  - andreyvelich
  - dhirajsb
  - tarilabs
  - Tomcli
reviewers:
  - andreyvelich
  - ckadner
  - dhirajsb
  - isinyaaa
  - lampajr
  - rareddy
  - rkubis
  - tarilabs
  - Tomcli
  - tonyxrmdavidson
```

README.md

```
# Model Registry
Model registry provides a central repository for model developers to store and manage models, versions, and artifacts metadata. A Go-based application that leverages [ml_metadata](https://github.com/google/ml-metadata/) project under the hood.

## Red Hat's Pledge
- Red Hat drives the project's development through Open Source principles, ensuring transparency, sustainability, and community ownership.
- Red Hat values the Kubeflow community and commits to providing a minimum of 12 months' notice before ending project maintenance after the initial release.
```

**Comment by zijianjoy** [comment]
@rareddy Please only select CNCF allowlist license in https://github.com/cncf/foundation/blob/main/allowed-third-party-license-policy.md#approved-licenses-for-allowlist.

**Comment by james-jwu** [comment]
Can we use Apache 2.0, which the majority of Kubeflow projects are using?

**Comment by rareddy** [comment]
@zijianjoy @james-jwu the language in the document seems to indicate ASL 2.0 is acceptable, in any case I am going to ask for legal approval before we can switch licenses from ALS 2.0.

**Comment by zijianjoy** [comment]
1. Repo https://github.com/kubeflow/model-registry is created. 
1. OWNERS file is also created. This is a placeholder because it mimics current member of `wg-model-registry-lead` for now.
2. Make sure you add yourself to wg-model-registry-lead and OWNERS file later on, remove any person that is not supposed to be in the owner group.
3. `tide` is configured but let me know if anything doesn't work.
4. README file is created.

**Comment by rareddy** [comment]
Yay! A huge Thank You to all. @zijianjoy @james-jwu @jbottum

**Comment by jbottum** [comment]
great progress in 2024!
